test/logs.bats

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bats
+# vim:set ft=bash :
+
+load helpers
+
+function setup() {
+	setup_test
+}
+
+function teardown() {
+	cleanup_test
+}
+
+@test "ReopenContainerLog should succeed when the container is running" {
+	start_crio
+
+	jq --arg image "$IMAGE" '.metadata.name = "sleep"
+		| .command = ["/bin/sh", "-c", "sleep 600"]' \
+		"$TESTDATA"/container_config.json > "$TESTDIR"/trap.json
+
+	ctr_id=$(crictl run "$TESTDIR"/trap.json "$TESTDATA"/sandbox_config.json)
+	crictl logs -r "$ctr_id"
+}
+
+@test "ReopenContainerLog should fail when the container is stopped" {
+	start_crio
+
+	jq --arg image "$IMAGE" '.metadata.name = "sleep"
+		| .command = ["/bin/sh", "-c", "sleep 600"]' \
+		"$TESTDATA"/container_config.json > "$TESTDIR"/trap.json
+
+	ctr_id=$(crictl run "$TESTDIR"/trap.json "$TESTDATA"/sandbox_config.json)
+	crictl stop "$ctr_id"
+	run ! crictl logs -r "$ctr_id"
```

Comment:
nit: no need for `run` if you're using `!` and expect this to fail

Reply:
https://bats-core.readthedocs.io/en/stable/writing-tests.html
> Note: In contrast to the above, testing that a command failed is best done via
> run ! command args ...

it's on the last line so it can be omitted though...

Reply:
TIL! FYI @kolyshkin I feel like years ago you recommended we drop `run` in these cases